src/features/user-profile-public/hooks/useReport.tsx

```diff
@@ -0,0 +1,24 @@
+import { useMutation } from "@tanstack/react-query";
+import { CreateReportDto, ReportTargetType, submitReport } from "../api/report.api";
+
+interface ReportVariables {
+    targetId: number;
+    targetType: ReportTargetType;
+    reason: string;
+}
+  
+export function useReport() {
+  return useMutation({
+    mutationFn: ({ targetId, reason, targetType }: ReportVariables) => {
+      const url = window.location.href;
```

Comment:
this seems wrong. If url is assigned like this, 
![image](https://github.com/user-attachments/assets/6ca6f98b-f194-4353-8227-1161a7880c83)
if we go to hosted environment like in vercel and trying to open the report, it will cause error. Please replace this with VITE_FE_URL
![image](https://github.com/user-attachments/assets/4790cdc1-5973-41ac-a891-2af3772b348d)

Reply:
Do the same thing for report post